[2.0.x] Kossel Pro (BRAINWAVE_PRO) example compile fail (Z_MIN_PIN)

Found while trying to fix Travis builds:

```
$ use_example_configs delta/kossel_pro
The command "use_example_configs delta/kossel_pro" exited with 0.
3.95s$ build_marlin_pio ${TRAVIS_BUILD_DIR} ${TEST_PLATFORM}
In file included from Marlin/src/HAL/HAL_AVR/../../inc/MarlinConfig.h:34:0,
from Marlin/src/HAL/HAL_AVR/HAL_spi_AVR.cpp:40:
Marlin/src/HAL/HAL_AVR/../../inc/SanityCheck.h:691:8: error: #error "Z_MIN_PROBE_USES_Z_MIN_ENDSTOP_PIN requires the Z_MIN_PIN to be defined."
#error "Z_MIN_PROBE_USES_Z_MIN_ENDSTOP_PIN requires the Z_MIN_PIN to be defined."
^
In file included from Marlin/src/HAL/HAL_AVR/../../inc/MarlinConfig.h:34:0,
from Marlin/src/HAL/HAL_AVR/MarlinSerial.cpp:36:
Marlin/src/HAL/HAL_AVR/../../inc/SanityCheck.h:691:8: error: #error "Z_MIN_PROBE_USES_Z_MIN_ENDSTOP_PIN requires the Z_MIN_PIN to be defined."
#error "Z_MIN_PROBE_USES_Z_MIN_ENDSTOP_PIN requires the Z_MIN_PIN to be defined."
^
In file included from Marlin/src/HAL/HAL_AVR/../../inc/MarlinConfig.h:34:0,
from Marlin/src/HAL/HAL_AVR/persistent_store_impl.cpp:5:
Marlin/src/HAL/HAL_AVR/../../inc/SanityCheck.h:691:8: error: #error "Z_MIN_PROBE_USES_Z_MIN_ENDSTOP_PIN requires the Z_MIN_PIN to be defined."
#error "Z_MIN_PROBE_USES_Z_MIN_ENDSTOP_PIN requires the Z_MIN_PIN to be defined."
^
In file included from Marlin/src/HAL/HAL_AVR/../../inc/MarlinConfig.h:34:0,
from Marlin/src/HAL/HAL_AVR/servo_AVR.cpp:56:
Marlin/src/HAL/HAL_AVR/../../inc/SanityCheck.h:691:8: error: #error "Z_MIN_PROBE_USES_Z_MIN_ENDSTOP_PIN requires the Z_MIN_PIN to be defined."
#error "Z_MIN_PROBE_USES_Z_MIN_ENDSTOP_PIN requires the Z_MIN_PIN to be defined."
^
In file included from Marlin/src/HAL/HAL_AVR/../../inc/MarlinConfig.h:34:0,
from Marlin/src/HAL/HAL_AVR/watchdog_AVR.cpp:25:
Marlin/src/HAL/HAL_AVR/../../inc/SanityCheck.h:691:8: error: #error "Z_MIN_PROBE_USES_Z_MIN_ENDSTOP_PIN requires the Z_MIN_PIN to be defined."
#error "Z_MIN_PROBE_USES_Z_MIN_ENDSTOP_PIN requires the Z_MIN_PIN to be defined."
^
*** [.pioenvs/teensy20/src/src/HAL/HAL_AVR/HAL_spi_AVR.o] Error 1
*** [.pioenvs/teensy20/src/src/HAL/HAL_AVR/MarlinSerial.o] Error 1
*** [.pioenvs/teensy20/src/src/HAL/HAL_AVR/persistent_store_impl.o] Error 1
*** [.pioenvs/teensy20/src/src/HAL/HAL_AVR/servo_AVR.o] Error 1
*** [.pioenvs/teensy20/src/src/HAL/HAL_AVR/watchdog_AVR.o] Error 1
In file included from Marlin/src/HAL/HAL_LPC1768/../../inc/MarlinConfig.h:34:0,
from Marlin/src/HAL/HAL_LPC1768/LPC1768_Servo.cpp:63:
Marlin/src/HAL/HAL_LPC1768/../../inc/SanityCheck.h:691:8: error: #error "Z_MIN_PROBE_USES_Z_MIN_ENDSTOP_PIN requires the Z_MIN_PIN to be defined."
#error "Z_MIN_PROBE_USES_Z_MIN_ENDSTOP_PIN requires the Z_MIN_PIN to be defined."
^
*** [.pioenvs/teensy20/src/src/HAL/HAL_LPC1768/LPC1768_Servo.o] Error 1
========================== [ERROR] Took 3.74 seconds ==========================
================================== [SUMMARY] ==================================
Environment megaatmega2560          	[SKIP]
Environment megaatmega1280          	[SKIP]
Environment anet10                  	[SKIP]
Environment DUE                     	[SKIP]
Environment LPC1768                 	[SKIP]
Environment LPC1768_debug_and_upload	[SKIP]
Environment rambo                   	[SKIP]
Environment sanguino_atmega644p     	[SKIP]
Environment STM32F1                 	[SKIP]
Environment teensy20                	[ERROR]
Environment teensy35                	[SKIP]
========================== [ERROR] Took 3.74 seconds ==========================
```

From readme originally created by @Wackerbarth :

> # Example Configuration for OpenBeam Kossel Pro [BRAINWAVE_PRO](http://www.openbeamusa.com/3d-printers/kossel/)
> * Configuration files for the **Openbeam Kossel Pro** as delivered in their KickStarter distribution
> 
> I [@Wackerbarth](https://github.com/Wackerbarth) tested this version on my Kossel Pro and Arduino 1.6.5 for Mac.
> This configuration is a transition to merge Terence Tam's configuration with up-to-date Marlin source and a current Arduino IDE

-=dave

Dave,

This configuration has been passed down from years ago. At that time, all of the processor/pins stuff was quite different. I upgraded the processor mine a long time ago and haven’t used Marlin on any delta printer in some time.

Perhaps the configuration should just “disappear” unless there are others interested in keeping it active. (The same can be said for other systems also. At some time, we need to call “END OF LIFE” on the dinosaurs)

> On Dec 15, 2017, at 2:49 PM, Dave Johnson <notifications@github.com> wrote:
> 
> Found while trying to fix Travis builds:
> 
> $ use_example_configs delta/kossel_pro
> The command "use_example_configs delta/kossel_pro" exited with 0.
> 3.95s$ build_marlin_pio ${TRAVIS_BUILD_DIR} ${TEST_PLATFORM}
> In file included from Marlin/src/HAL/HAL_AVR/../../inc/MarlinConfig.h:34:0,
> from Marlin/src/HAL/HAL_AVR/HAL_spi_AVR.cpp:40:
> Marlin/src/HAL/HAL_AVR/../../inc/SanityCheck.h:691:8: error: #error "Z_MIN_PROBE_USES_Z_MIN_ENDSTOP_PIN requires the Z_MIN_PIN to be defined."
> #error "Z_MIN_PROBE_USES_Z_MIN_ENDSTOP_PIN requires the Z_MIN_PIN to be defined."
> ^
> In file included from Marlin/src/HAL/HAL_AVR/../../inc/MarlinConfig.h:34:0,
> from Marlin/src/HAL/HAL_AVR/MarlinSerial.cpp:36:
> Marlin/src/HAL/HAL_AVR/../../inc/SanityCheck.h:691:8: error: #error "Z_MIN_PROBE_USES_Z_MIN_ENDSTOP_PIN requires the Z_MIN_PIN to be defined."
> #error "Z_MIN_PROBE_USES_Z_MIN_ENDSTOP_PIN requires the Z_MIN_PIN to be defined."
> ^
> In file included from Marlin/src/HAL/HAL_AVR/../../inc/MarlinConfig.h:34:0,
> from Marlin/src/HAL/HAL_AVR/persistent_store_impl.cpp:5:
> Marlin/src/HAL/HAL_AVR/../../inc/SanityCheck.h:691:8: error: #error "Z_MIN_PROBE_USES_Z_MIN_ENDSTOP_PIN requires the Z_MIN_PIN to be defined."
> #error "Z_MIN_PROBE_USES_Z_MIN_ENDSTOP_PIN requires the Z_MIN_PIN to be defined."
> ^
> In file included from Marlin/src/HAL/HAL_AVR/../../inc/MarlinConfig.h:34:0,
> from Marlin/src/HAL/HAL_AVR/servo_AVR.cpp:56:
> Marlin/src/HAL/HAL_AVR/../../inc/SanityCheck.h:691:8: error: #error "Z_MIN_PROBE_USES_Z_MIN_ENDSTOP_PIN requires the Z_MIN_PIN to be defined."
> #error "Z_MIN_PROBE_USES_Z_MIN_ENDSTOP_PIN requires the Z_MIN_PIN to be defined."
> ^
> In file included from Marlin/src/HAL/HAL_AVR/../../inc/MarlinConfig.h:34:0,
> from Marlin/src/HAL/HAL_AVR/watchdog_AVR.cpp:25:
> Marlin/src/HAL/HAL_AVR/../../inc/SanityCheck.h:691:8: error: #error "Z_MIN_PROBE_USES_Z_MIN_ENDSTOP_PIN requires the Z_MIN_PIN to be defined."
> #error "Z_MIN_PROBE_USES_Z_MIN_ENDSTOP_PIN requires the Z_MIN_PIN to be defined."
> ^
> *** [.pioenvs/teensy20/src/src/HAL/HAL_AVR/HAL_spi_AVR.o] Error 1
> *** [.pioenvs/teensy20/src/src/HAL/HAL_AVR/MarlinSerial.o] Error 1
> *** [.pioenvs/teensy20/src/src/HAL/HAL_AVR/persistent_store_impl.o] Error 1
> *** [.pioenvs/teensy20/src/src/HAL/HAL_AVR/servo_AVR.o] Error 1
> *** [.pioenvs/teensy20/src/src/HAL/HAL_AVR/watchdog_AVR.o] Error 1
> In file included from Marlin/src/HAL/HAL_LPC1768/../../inc/MarlinConfig.h:34:0,
> from Marlin/src/HAL/HAL_LPC1768/LPC1768_Servo.cpp:63:
> Marlin/src/HAL/HAL_LPC1768/../../inc/SanityCheck.h:691:8: error: #error "Z_MIN_PROBE_USES_Z_MIN_ENDSTOP_PIN requires the Z_MIN_PIN to be defined."
> #error "Z_MIN_PROBE_USES_Z_MIN_ENDSTOP_PIN requires the Z_MIN_PIN to be defined."
> ^
> *** [.pioenvs/teensy20/src/src/HAL/HAL_LPC1768/LPC1768_Servo.o] Error 1
> ========================== [ERROR] Took 3.74 seconds ==========================
> ================================== [SUMMARY] ==================================
> Environment megaatmega2560          	[SKIP]
> Environment megaatmega1280          	[SKIP]
> Environment anet10                  	[SKIP]
> Environment DUE                     	[SKIP]
> Environment LPC1768                 	[SKIP]
> Environment LPC1768_debug_and_upload	[SKIP]
> Environment rambo                   	[SKIP]
> Environment sanguino_atmega644p     	[SKIP]
> Environment STM32F1                 	[SKIP]
> Environment teensy20                	[ERROR]
> Environment teensy35                	[SKIP]
> ========================== [ERROR] Took 3.74 seconds ==========================
> From readme originally created by @Wackerbarth <https://github.com/wackerbarth> :
> 
> Example Configuration for OpenBeam Kossel Pro BRAINWAVE_PRO <http://www.openbeamusa.com/3d-printers/kossel/>
> Configuration files for the Openbeam Kossel Pro as delivered in their KickStarter distribution
> I @Wackerbarth <https://github.com/Wackerbarth> tested this version on my Kossel Pro and Arduino 1.6.5 for Mac.
> This configuration is a transition to merge Terence Tam's configuration with up-to-date Marlin source and a current Arduino IDE
> 
> -=dave
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub <https://github.com/MarlinFirmware/Marlin/issues/8795>, or mute the thread <https://github.com/notifications/unsubscribe-auth/AAe160hbsL3dJQOCYtWoiQ8JTqV5iLYrks5tAttRgaJpZM4RD-9v>.
>

--- a/Marlin/src/config/examples/delta/kossel_pro/Configuration.h
+++ b/Marlin/src/config/examples/delta/kossel_pro/Configuration.h
@@ -671,7 +671,7 @@
  *
  * Enable this option for a probe connected to the Z Min endstop pin.
  */
-#define Z_MIN_PROBE_USES_Z_MIN_ENDSTOP_PIN
+//#define Z_MIN_PROBE_USES_Z_MIN_ENDSTOP_PIN
 
 /**
  * Z_MIN_PROBE_ENDSTOP
@@ -692,7 +692,7 @@
  * disastrous consequences. Use with caution and do your homework.
  *
  */
-//#define Z_MIN_PROBE_ENDSTOP
+#define Z_MIN_PROBE_ENDSTOP
 
 /**
  * Probe Type

--- a/Marlin/src/inc/SanityCheck.h
+++ b/Marlin/src/inc/SanityCheck.h
@@ -621,9 +621,7 @@ static_assert(1 >= 0
  * Delta requirements
  */
 #if ENABLED(DELTA)
-  #if HAS_BED_PROBE && ENABLED(Z_MIN_PROBE_ENDSTOP)
-    #error "Delta probably shouldn't use Z_MIN_PROBE_ENDSTOP. Comment out this line to continue."
-  #elif DISABLED(USE_XMAX_PLUG) && DISABLED(USE_YMAX_PLUG) && DISABLED(USE_ZMAX_PLUG)
+  #if DISABLED(USE_XMAX_PLUG) && DISABLED(USE_YMAX_PLUG) && DISABLED(USE_ZMAX_PLUG)
     #error "You probably want to use Max Endstops for DELTA!"
   #elif ENABLED(ENABLE_LEVELING_FADE_HEIGHT) && DISABLED(AUTO_BED_LEVELING_BILINEAR) && !UBL_SEGMENTED
     #error "ENABLE_LEVELING_FADE_HEIGHT on DELTA requires AUTO_BED_LEVELING_BILINEAR or AUTO_BED_LEVELING_UBL."

